src/pushover_complete/pushover_api.py

```diff
@@ -141,13 +142,17 @@ def _send_message(self, user, message, device=None, title=None, url=None, url_ti
             'html': html
         }
         
-        if image_path is not None and os.path.isfile(image_path):
-            with open(image_path, 'rb') as f:
-                file = {'attachment': f}
-                return self._generic_post('messages.json', payload=payload, session=session, files=file)
+        if image is not None:
+            if isinstance(image, file) or isinstance(image, io.BytesIO):
+                    attachment = {'attachment': image}
+                    return self._generic_post('messages.json', payload=payload, session=session, files=attachment)
+            if isinstance(image, str) and os.path.isfile(image):
```

Comment:
use `six.string_types` for python2/3 compatibility

Reply:
What part breaks? I tested this with python2. I'm looking in to `six` as well.

Reply:
Ah, str is only python 3? Strange that it worked for me. I'll add in six usage.